GTC-2622 Add a test for gfwpro_dashboard

danscales opened the pull request:
GTC-2622 Add test for gfwpro_dashboard

I added several input files, and had to change the return value of
GfwProDashboardAnalysis().

Also, I had to add ability to pin a version of a dataset (see second commit),
since we need to ping the integrated_alerts version so that the dashboard
results from the test are constant.

danscales committed:
GTC-2622 Add test for gfwpro_dashboard

To allow for the test, I needed to change GfwProDashboardAnalysis to
return the RDD, instead of doing the full export and returning nothing.
(Similar to ForestChangeDiagnosticAnalysis, which also returns an RDD so
that writing a test is possible.)

I added an input file dash.tsv for the test. I also added a file
idn1_5Gadm.tsv which is just the row from gadm36_adm2_1_1.tsv for
IDN1.5, which is the only area needed for the particular input geometry
in dash.tsv.

In this next commit, I will add a pinning option, so the output stays
constant even as new integrated alerts datasets are added.

danscales committed:
Add ability to pin a version for a dataset.

The pinned command-line option already exists, but it was no longer
functional, so I re-enabled it. To do that, I had to add a new 'pinned'
argument to GfwConfig.get(), so the 'pinned' value could control which
version is used from the catalog.

We are using it specifically for the gfwpro_dashboard test, where we
need to pin the integrated_alerts version so that the dashboard results
from the test are constant.

jterry64 commented on src/main/scala/org/globalforestwatch/summarystats/carbonflux/CarbonFluxCommand.scala:
```diff
@@ -20,7 +20,7 @@ object CarbonFluxCommand extends SummaryCommand {
         "noOutputPathSuffix" -> default.noOutputPathSuffix,
         "changeOnly" -> changeOnly,
-        "config" -> GfwConfig.get
+        "config" -> GfwConfig.get(None)
```
If you look in SummaryCommand, there's already a `pinnedVersions` input parameter to the geotrellis command. Should that be propagated instead? 

Also feels a little strange to have to pass None to each individual get function, I forget, does Scala not have default parameters?

danscales replied:
> Also feels a little strange to have to pass None to each individual get function, I forget, does Scala not have default > parameters?

Good suggestion about adding a default parameter for GfwConfig.get().  I've made that change.  Since GfwConfig.get does have an argument now, you do have use `GfwConfig.get()` everywhere now, rather than just `GfwConfig.get`.

> If you look in SummaryCommand, there's already a pinnedVersions input parameter to the geotrellis command.
> Should that be propagated instead?

In the new GfwProDashboardAnalysisSpec test, I specify a specific pinned version for the test, so of course, I don't need or use pinnedVersionsOpts (and the tests don't have access to it anyway)

I changed the code in GfwProDashboardCommand() to use pinnedVersionOpts as an example.  I doubt that any submitted analysis command will want to use pinned versions (other than for tests), so I didn't change all the other analyses, but let me know if you want me to do that.  Pinning only applies to the data sets that use `latest` as their version, which currently are only glad alerts and integrated alerts.

danscales committed:
Change GfwConfig.get() to have default arg of None (no pinning)

And I added an example of using the pinnedVersionsOpts command-line arg
to control pinning in the gfw_pro_dashboard analysis. I would really
expected we would use pinning of integrated alerts or glad alerts in
anything other than tests (which is what I am doing in the
GfwProDashboardAnalysisSpec test.

jterry64 reviewed (approved):
Thanks for making the changes!

danscales commented:
Thanks for the review and re-review!
